### ISSUE TITLE
500 error on upgrade from 5.somthing to 6.3

### ISSUE THREAD

**Issue by sgtwilko** [jakubgarfield/Bonobo-Git-Server#756]
Hi,

I've been trying to upgrade from 5 to 6.3.

I keep getting the error shown below and at this point the ```app_data\ADBackend``` folder is almost wiped out.

I've checked the permissions, I've installed .Net 4.6.2, I'm at a loss as to how to continue to fix this.

Error shown on screen:
````
Server Error in '/Bonobo.Git.Server' Application.
--------------------------------------------------------------------------------

Runtime Error 
Description: An exception occurred while processing your request. Additionally, another exception occurred while executing the custom error page for the first exception. The request has been terminated. 
````

Here is the error from the log:
```
2017-11-06 18:12:56.998 +00:00 [Information] Searching for group "Redacted_group" in domain "redacted.domain.com"
2017-11-06 18:12:57.011 +00:00 [Error] Startup exception
System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Bonobo.Git.Server.Data.Update.ADBackendUpdate.Pre600UpdateTo600.UpdateTeams(String dir, Dictionary`2 users)
   at Bonobo.Git.Server.Data.Update.ADBackendUpdate.Pre600UpdateTo600.UpdateADBackend()
   at Bonobo.Git.Server.MvcApplication.Application_Start()
```

**Comment by larshg** [comment]
It seems it fail updating the teams directly from your version to the newest. 

I can't remember if you need to update one version to the next.

If you want a new 6.something I would go for 6.2.1 because there are something not working with 6.3 with AD. See #755.

The only way is probably to set up a empty new instance of bonobo and the copy only the repositories which would end up as 'found in filesystem' but nothing else of information, team and personal assignment etc.

**Comment by sgtwilko** [comment]
@larshg Thanks for the quick response.
I managed to get a bit further by myself.

I found a couple of issues that stopped the upgrade.

Firstly, in the team members list we had a user ```null````.
Secondly, at one time we had a sync issue and had two users with the same username in AD.

This hadn't been an issue until the upgrade tried to map the two users to the same UID.

After all that I couldn't log in, but downgrading to 6.2.1 sorted the AD login.

Thanks for your help.
